### PR TITLE
added solution for kattis-pointpolygon

### PR DESCRIPTION
**Pull request by RedBlueBird**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/kattis-pointinpolygon.mdx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+const int MAX = 1e5+5;
+
+struct point{
+    ll x,y;
+};
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:60]: rename to Point

File: solutions/kattis-pointinpolygon.mdx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+                    cout << "on\n";
+                    intersections = -1;
+                    break;
+                }
+            }
+            if (intersections != -1) {
+                if (intersections % 2)
+                    cout << "in\n";
+                else
+                    cout << "out\n";
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:133]: braces

File: solutions/kattis-pointinpolygon.mdx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            break;
+        }
+        for (int i = 0; i < n; i++) {
+            cin >> polygon[i].x >> polygon[i].y;
+        }
+        cin >> m;
+
+        //Cast an ray and count # line segments that the ray intersected with:
+        //  If the amount is odd, then the point is inside, otherwise outside
+        //  If the point is on a line segment, then say it is on the boundary
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:113]:
```cpp
/*
 * multiline comments like this
 * don't do a bunch of single lines
 */
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/kattis-pointinpolygon.mdx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                    cout << "on\n";
+                    intersections = -1;
+                    break;
+                }
+            }
+            if (intersections != -1) {
+                if (intersections % 2){
+                    cout << "in\n";
+                }
+                else{
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:139]:
```suggestion
                } else {
```

File: solutions/kattis-pointinpolygon.mdx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+bool intersect(Point p1, Point p2, Point p3, Point p4){
+    return ((dir(p1,p2,p3)!=dir(p1,p2,p4))&&(dir(p3,p4,p1)!=dir(p3,p4,p2)));
+}
+
+/*
+ * If point p3 is collinear with points p1 and p2
+ * Check if p3 is inside or outside the line segment (p1,p2)
+ */
+bool within(Point p1, Point p2, Point p3){
+    if (dir(p1,p2,p3)==0){
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:90]: tabs to indent, it's the convention around here
also use spaces around operators everywhere

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/kattis-pointinpolygon.mdx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+const int MAX = 1e5+5;
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:56]: is the +5 needed?

**Comment by RedBlueBird** [comment]
Is there anything else that I need to change

File: solutions/kattis-pointinpolygon.mdx [hunk 7]
```diff
@@ -53,7 +53,7 @@ As all geometry problem go, edge cases are a must. One edge case one might consi
 #include <bits/stdc++.h>
 using namespace std;
 typedef long long ll;
-const int MAX = 1e5+5;
+const int MAX = 1e4+1;
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:56]: again, is the +1 needed lol

Comment (RedBlueBird) [reply solutions/kattis-pointinpolygon.mdx:56]: hold up, you are actually right

File: solutions/kattis-pointinpolygon.mdx [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+        bool xRange = (min(p1.x, p2.x) <= p3.x) && (p3.x <= max(p1.x, p2.x));
+        bool yRange = (min(p1.y, p2.y) <= p3.y) && (p3.y <= max(p1.y, p2.y));
+        return xRange && yRange;
+    } else {
+        return false;
+    }
+}
+
+Point polygon[MAX];
+int n, m;
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:101]: (optional) use vector
(please do this) put `n` and `m` inside the while loop of `main`

**Review by SansPapyrus683 — APPROVED** [review]
thanks for putting up with me jeff

File: solutions/kattis-pointinpolygon.mdx [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+
+struct Point{
+    ll x, y;
```
Comment (dongliu0426) [inline solutions/kattis-pointinpolygon.mdx:58]: use tabs instead of spaces for indentation (same goes for below)

**Review by dongliu0426 — APPROVED** [review]

File: solutions/kattis-pointinpolygon.mdx [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+$$
+\vec{b_1} = p_3-p_2
+$$
+$$
+\vec{b_2} = p_4-p_2
+$$
+
+The condition will be true if and only if $\vec{a}\times\vec{b_1} \neq \vec{a}\times\vec{b_2}$. Verifying the second of the two cases is equivalent to the procedure above, except $p_1$ and $p_2$ are switched with $p_3$ and $p_4$.
+
+As all geometry problem go, edge cases are a must. One edge case one might consider in our method above is that if both $\vec{a}\times\vec{b_1}$ and $\vec{a}\times\vec{b_2}$ are equal and are equal to 0, then they can still intersect as long as they overlap with each other. But let me in you a ***secret***, due to the way this problem is structured, all vertices are on lattice points, meaning that they have whole numbers as coordinates. And since one of the $L$ is customizable (we get to pick the point the point cast its ray at), we can just set its coordinate to $(\infty,1)$. This way, the ray will have a slope of close to $0$, successfully avoiding all possible lattice point intersections. One trick to rule edges cases all!
```
Comment (nwatx) [inline solutions/kattis-pointinpolygon.mdx:44]: > but let me in you a secret

did you mean but let me let you in on a secret?

Comment (RedBlueBird) [reply solutions/kattis-pointinpolygon.mdx:44]: oops, you are right! Thanks for catching the mistake haha.

**Comment by RedBlueBird** [comment]
Wait a minute, I just realized some reason the Vercel bot is not deploying a review this time.

**Review by ryanchou-dev — APPROVED** [review]
happy first pr!!

File: solutions/kattis-pointinpolygon.mdx [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+$$
+\vec{b_1} = p_3-p_2
+$$
+$$
+\vec{b_2} = p_4-p_2
+$$
+
+The condition will be true if and only if $\vec{a}\times\vec{b_1} \neq \vec{a}\times\vec{b_2}$. Verifying the second of the two cases is equivalent to the procedure above, except $p_1$ and $p_2$ are switched with $p_3$ and $p_4$.
+
+As all geometry problem go, edge cases are a must. One edge case one might consider in our method above is that if both $\vec{a}\times\vec{b_1}$ and $\vec{a}\times\vec{b_2}$ are equal and are equal to 0, then they can still intersect as long as they overlap with each other. But let me let you in on a ***secret***, due to the way this problem is structured, all vertices are on lattice points, meaning that they have whole numbers as coordinates. And since one of the $L$ is customizable (we get to pick the point the point cast its ray at), we can just set its coordinate to $(\infty,1)$. This way, the ray will have a slope of close to $0$, successfully avoiding all possible lattice point intersections. One trick to rule edges cases all!
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:43]: > One trick to rule edge cases all!

One trick to rule all the edge cases!

File: solutions/kattis-pointinpolygon.mdx [hunk 12]
```diff
@@ -0,0 +1,144 @@
+---
+id: kattis-pointinpolygon
+source: Kattis
+title: Point in Polygon
+author: Jeffrey Zhang
+---
+
+## Solution 
+In order to determine if a point is inside a polygon, we can imagine a ray being casted from that point towards any direction. Resulting in three possible scenarios:
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:9]: "cast" instead of "casted"

File: solutions/kattis-pointinpolygon.mdx [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+source: Kattis
+title: Point in Polygon
+author: Jeffrey Zhang
+---
+
+## Solution 
+In order to determine if a point is inside a polygon, we can imagine a ray being casted from that point towards any direction. Resulting in three possible scenarios: 
+
+1. If the ray intersects with odd number of polygon edges, then the point is inside.
+2. If the ray intersects with even number of polygon edges, then the point is outside.
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:12]: just "edges", not "polygon edges"

File: solutions/kattis-pointinpolygon.mdx [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+3. If the origin point is on a line segment, then the point is on the edge.
+
+This algorithm is known as the [*Ray Casting Algorithm*](https://en.wikipedia.org/wiki/Point_in_polygon#Ray_casting_algorithm), more details can be found [here](https://en.wikipedia.org/wiki/Point_in_polygon)
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:15]: the more details part seems unnecessary

File: solutions/kattis-pointinpolygon.mdx [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+
+## Line segment & Line segment intersection
+
+Given two line segments $L_1 = (p_1,p_2)$ and $L_2 = (p_3,p_4)$, how can we effectively check if they intersect each other? We will first have to use a concept from linear algebra called the *Determinant*.
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:19]: efficiently, not effectively
also don't capitalize determinant

File: solutions/kattis-pointinpolygon.mdx [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+
+<Info title="Complementary resource">
+
+If you don't already know what a determinant is, watch this [youtube video](https://www.youtube.com/watch?v=Ip3X9LOh2dk) for an in-depth explanation.
+
+</Info>
+
+We will define the determinant operator of two vectors as $\times$. Given two vectors $\vec{a}=\langle x_1,y_1 \rangle$ and $\vec{b}=\langle x_2,y_2 \rangle$. If $\vec{a}\times\vec{b}$ gives a positive area then $\vec{b}$ points to the left of $\vec{a}$, if it gives a negative area then $\vec{b}$ points to the right of $\vec{a}$, and if it gives a zero area then $\vec{b}$ and $\vec{a}$ are parallel.
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:27]: determinant, not determinant operator

File: solutions/kattis-pointinpolygon.mdx [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+
+This algorithm is known as the [*Ray Casting Algorithm*](https://en.wikipedia.org/wiki/Point_in_polygon#Ray_casting_algorithm), more details can be found [here](https://en.wikipedia.org/wiki/Point_in_polygon)
+
+## Line segment & Line segment intersection
+
+Given two line segments $L_1 = (p_1,p_2)$ and $L_2 = (p_3,p_4)$, how can we effectively check if they intersect each other? We will first have to use a concept from linear algebra called the *Determinant*.
+
+<Info title="Complementary resource">
+
+If you don't already know what a determinant is, watch this [youtube video](https://www.youtube.com/watch?v=Ip3X9LOh2dk) for an in-depth explanation.
```
Comment (SansPapyrus683) [inline solutions/kattis-pointinpolygon.mdx:23]: This YouTube video provides a handy in-depth explanation of determinants.

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
sorry jeff, found some grammar issues

**Review by SansPapyrus683 — APPROVED** [review]
yee

**Pull request merged**